core/src/test/scala/org/apache/spark/crypto/CryptoCodecSuite.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io._
+import java.security.SecureRandom
+import java.util
+import java.util.Random
+
+import com.intel.chimera._
+import com.intel.chimera.codec._
+
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.crypto.CommonConfigurationKeys._
+import org.apache.spark.crypto.CryptoStreamUtils._
+
+private[spark] class CryptoCodecSuite extends SparkFunSuite {
```

Comment:
As I mentioned before, none of this is testing Spark code. These tests don't need to exist in Spark at all; we assume this has been tested by Chimera itself.

What you need are explicit tests for the code you're adding in `CryptoConf` and `CryptoStreamUtils`. e.g., a test that makes sure that `toChimeraConf` works (and right now it doesn't because it's checking the wrong thing!).